packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts

```diff
@@ -0,0 +1,745 @@
+      if (observer) {
+        observer.next({
+          data: payload,
+          extensions: {
+            controlMsgType: "CONNECTED"
+          }
+        });
+      } else {
+        logger(`observer not found for id: ${id}`);
+      }
+
+      const subscriptionState = SUBSCRIPTION_STATUS.CONNECTED;
+      this.subscriptionObserverMap.set(id, {
+        observer,
+        query,
+        variables,
+        startAckTimeoutId: null,
+        subscriptionState,
+        subscriptionReadyCallback,
+        subscriptionFailedCallback
+      });
+
+      return;
+    }
+
+    if (type === MESSAGE_TYPES.GQL_CONNECTION_KEEP_ALIVE) {
+      clearTimeout(this.keepAliveTimeoutId);
+      this.keepAliveTimeoutId = setTimeout(
+        this._timeoutDisconnect.bind(this),
+        this.keepAliveTimeout
+      );
+      return;
+    }
+
+    if (type === MESSAGE_TYPES.GQL_ERROR) {
+      const subscriptionState = SUBSCRIPTION_STATUS.FAILED;
+      this.subscriptionObserverMap.set(id, {
+        observer,
+        query,
+        variables,
+        startAckTimeoutId,
+        subscriptionReadyCallback,
+        subscriptionFailedCallback,
+        subscriptionState
+      });
+
+      observer.error({
+        errors: [
+          {
+            ...new GraphQLError(`Connection failed: ${JSON.stringify(payload)}`)
+          }
+        ]
+      });
+      clearTimeout(startAckTimeoutId);
+
+      observer.complete();
+      if (typeof subscriptionFailedCallback === "function") {
+        subscriptionFailedCallback();
+      }
+    }
+  }
+
+  private _timeoutDisconnect() {
+    this.subscriptionObserverMap.forEach(({ observer }) => {
+      observer.error({
+        errors: [{ ...new GraphQLError(`Timeout disconnect`) }]
+      });
+      observer.complete();
+    });
+    this.subscriptionObserverMap = new Map();
+    if (this.awsRealTimeSocket) {
+      this.awsRealTimeSocket.close();
+    }
+
+    this.socketStatus = SOCKET_STATUS.CLOSED;
+  }
+
+  private _timeoutStartSubscriptionAck(subscriptionId) {
+    const { observer, query, variables } = this.subscriptionObserverMap.get(
+      subscriptionId
+    );
+    this.subscriptionObserverMap.set(subscriptionId, {
+      observer,
+      query,
+      variables,
+      subscriptionState: SUBSCRIPTION_STATUS.FAILED
+    });
+
+    observer.error({
+      errors: [
+        {
+          ...new GraphQLError(
+            `Subscription timeout ${JSON.stringify({ query, variables })}`
+          )
+        }
+      ]
+    });
+    // Cleanup will be automatically executed
+    observer.complete();
+    logger("timeoutStartSubscription", JSON.stringify({ query, variables }));
+  }
+
+  static createWebSocket(awsRealTimeUrl: string, protocol: string): WebSocket {
+    return new WebSocket(awsRealTimeUrl, protocol);
+  }
+
+  private static _discoverAppSyncRealTimeEndpoint(url: string): string {
+    return url
+      .replace("https://", "wss://")
+      .replace("appsync-api", "appsync-realtime-api")
```

Comment:
Hi there,
What if we are using a custom domain name for appsync which doesn't include `appsync-api` in it. It will be the same domain name for realtime pure WebSocket, which doesn't work...